app/models/asset.rb

```diff
@@ -20,11 +20,22 @@ class Asset < ApplicationRecord
                       all: '-background "#d2d2d2" -flatten +matte'
                     }
 
+  has_attached_file :preview,
```

Comment:
Is there a specific reason why you moved preview to separate attachment?

Reply:
I'm not happy with it either, but I couldn't make it work with the PaperClip preprocessors (especially because it expects images and we can have double stuff going on in here, files/images) and I was already way over the estimated effort. Additionally I believe this might simplify migration to ActiveStorage when this will become redundant. How would you go about it?

Reply:
We can create custom post processor for paperclip
https://github.com/thoughtbot/paperclip#dynamic-configuration

Or simply set 'processing' flag on asset update, create delayed job for preview generation and save that preview directly to required style (Asset.first.file.path(:small)).